File: 프로그래머스/unrated/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> bandage, int health, vector<vector<int>> attacks) {
    int end = attacks[attacks.size() - 1][0];
    int maxHealth = health;
    int j = 0;
    int stroke = 0;
    for (int i = 1; i <= end; i++) {
        if (attacks[j][0] == i && j < attacks.size()) {
            health -= attacks[j++][1];
            stroke = 0;
            if (health <= 0)
                return -1;
        } else {
            health += bandage[1];
            stroke++;
            if (stroke == bandage[0]) {
                health += bandage[2];
                stroke = 0;
            }
            if (health > maxHealth)
                health = maxHealth;
        }
    }
    
    return health;
}

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int answer = 0, fever = 0;   
        int max = health;
        
        int t = bandage[0], x = bandage[1], y = bandage[2];
        
        if (attacks[0][0] == 0) health -= attacks[0][1];
        if (health <= 0) return -1;

        for (int i = 1, j = 0; i <= attacks[attacks.length - 1][0]; i++) {
            if (i == attacks[j][0]) {
                health -= attacks[j][1];
                if (health <= 0) return -1;
                
                fever = 0;
                j++;
            } else {
                health += x;
                fever++;

                if (fever == t) {
                    health += y;
                    fever = 0;
                }
                
                if (health > max) health = max;
            }
        }
        
        return health;
    }
}